packages/frameworks-solid-start/src/handler.ts

```diff
@@ -0,0 +1,132 @@
+import { AuthHandler, type AuthAction, type AuthOptions } from "@auth/core";
+import { Cookie, parseString, splitCookiesString } from "set-cookie-parser";
+import { serialize } from "cookie";
+
+export interface SoliduthOptions extends AuthOptions {
+  /**
+   * Defines the base path for the auth routes.
+   * @default '/api/auth'
+   */
+  prefix?: string;
+}
+
+const actions: AuthAction[] = [
+  "providers",
+  "session",
+  "csrf",
+  "signin",
+  "signout",
+  "callback",
+  "verify-request",
+  "error",
+  "_log",
+];
+
+export type ISolidAuthHandlerOpts = AuthOptions & {
+  failureRedirect?: string;
+};
+
+// currently multiple cookies are not supported, so we keep the next-auth.pkce.code_verifier cookie for now:
+// because it gets updated anyways
+// src: https://github.com/solidjs/solid-start/issues/293
+const getSetCookieCallback = (cook?: string | null): Cookie | undefined => {
+  if (!cook) return;
+  const splitCookie = splitCookiesString(cook);
+  for (const cookName of [
+    "__Secure-next-auth.session-token",
+    "next-auth.session-token",
+    "next-auth.pkce.code_verifier",
+    "__Secure-next-auth.pkce.code_verifier",
+  ]) {
+    const temp = splitCookie.find((e) => e.startsWith(`${cookName}=`));
+    if (temp) {
+      return parseString(temp);
+    }
+  }
+  return parseString(splitCookie?.[0] ?? ""); // just return the first cookie if no session token is found
+};
+
+function SolidAuthHandler(prefix: string, authOptions: ISolidAuthHandlerOpts) {
+  return async (event: APIEvent) => {
+    const { request } = event;
+    const url = new URL(request.url);
+    const [action] = url.pathname.slice(prefix.length + 1).split("/");
+    if (
+      actions.includes(action as AuthAction) &&
+      url.pathname.startsWith(prefix + "/")
+    ) {
+      const res = await AuthHandler(request, authOptions);
+      if (["callback", "signin", "signout"].includes(action)) {
+        const parsedCookie = getSetCookieCallback(
+          res.clone().headers.get("Set-Cookie")
+        );
+        if (parsedCookie) {
+          res.headers.set(
+            "Set-Cookie",
+            serialize(
+              parsedCookie.name,
+              parsedCookie.value,
+              parsedCookie as any
+            )
+          );
+        }
+      }
+      return res;
+    }
+    const h = authOptions.failureRedirect ?? "/";
+    const error = url.searchParams.get("error");
+    const error_description = url.searchParams.get("error_description");
+    const error_uri = url.searchParams.get("error_uri");
+    throw redirect(
+      `${h}?error=${error}&error_description=${error_description}&error_uri=${error_uri}`,
+      {
+        status: 302,
+      }
+    );
+  };
```

Comment:
This should not be in this package, should be handled in core instead.

I have a PR that already touches this: https://github.com/nextauthjs/next-auth/pull/6085/files#diff-70d010ea0fa6560101fc54c0385627d0937fbaca593c7df26a3c59729ac941d8R80-R87

Reply:
so i will remove it for now. let me know when its published